tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Framework.cs

```diff
@@ -221,31 +221,32 @@ internal void CheckAndBlock(Dictionary<string, object> args)
         {
             var reporting = MakeReportingFunction(result.Data, result.AggregatedTotalRuntime, result.AggregatedTotalRuntimeWithBindings);
 
-            var blocked = false;
+            BlockingAction? blockingAction = null;
             if (result.ShouldBlock)
             {
-                blocked = ChooseBlockingMethodAndBlock(result.Actions[0], reporting);
+                blockingAction = ChooseBlockingMethodAndBlock(result.Actions[0], reporting);
             }
 
-            reporting(blocked);
+            // here we assume if the we haven't blocked we'll have collected the correct status elsewhere
+            reporting(blockingAction?.StatusCode, result.ShouldBlock);
```

Comment:
I've moved the blocking before the collection of the status code. I didn't do it at first, as I thought it would have a big impact, but in the end, it's seems to a fairly small change. See changes in 0ce2bcc51

We still need to collect the status when throw.